autopush/db.py

```diff
@@ -658,16 +841,19 @@ def drop_user(self, uaid):
         """Drops a user record"""
         # The following hack ensures that only uaids that exist and are
         # deleted return true.
-        huaid = hasher(uaid)
-        return self.table.delete_item(uaid=huaid,
-                                      expected={"uaid__eq": huaid})
+        try:
+            self.get_uaid(uaid)
```

Comment:
Why add the get_uaid call here? It could be dangerous as get_uaid potentially calls drop_user

Reply:
Pretty much for the reason noted. We only delete UAIDs that already exist. If get_uaid deletes the uaid, then it was already scheduled for deletion and returning False is acceptable. Either way, it's deleted.

Reply:
This can turn into infinite recursion w/ an invalid record, though. See the case in test_db.test_incomplete_uaid -- if you disable the mock of drop_user in there it'll blow up the stack